### PR TITLE
Feature/peer connection

### PR DESCRIPTION
**Pull request by amanjaiswalofficial**

### CODE REVIEW

File: ops/handshake/message.go [hunk 1]
```diff
@@ -19,7 +27,15 @@ type Message map[string]*attrib
...
-	message[keepAlive] = &attrib{prefix: []byte{0,0,0,0}, messageID: 1}
+	message[keepAlive] = &attrib{prefix: []byte{0,0,0,0}, messageID: 0}
+	message[choke] = &attrib{prefix: []byte{0,0,0,1}, messageID: 0}
+	message[unchoke] = &attrib{prefix: []byte{0,0,0,1}, messageID: 1}
+	message[interested] = &attrib{prefix: []byte{0,0,0,1}, messageID: 2}
+	message[notIntersted] = &attrib{prefix: []byte{0,0,0,1}, messageID: 3}
+	message[have] = &attrib{prefix: []byte{0,0,0,5}, messageID: 5}
+	message[request] = &attrib{prefix: []byte{0,0,1,3}, messageID:6}
+	message[piece] = &attrib{prefix: []byte{0,0,0,9}, messageID:7}
+	message[cancel] = &attrib{prefix: []byte{0,0,1,3}, messageID:8}
```
Comment (ghost) [inline ops/handshake/message.go]: I don't quite understand what those byte slices means. I got the part of the messages id, but why some bytes slices are zeroes and others are set ???

File: ops/handshake/message.go [hunk 2]
```diff
@@ -1,52 +1,35 @@
...
+// convert the message into a byteslice to be sent to peers
+// based on the key provided
+func (msg Message) serializeMessage(prefixLen messageID) []byte {
 
-	return message
+	data := make([]byte, prefixLen+4)
 
-}
+	binary.BigEndian.PutUint32(data[0:4], uint32(prefixLen))
+	data[4] = byte(prefixLen)
```
Comment (ghost) [inline ops/handshake/message.go:33]: The payload is missing.

File: ops/handshake/message.go [hunk 3]
```diff
@@ -1,52 +1,35 @@
...
+	piece        messageID = 7
+	cancel       messageID = 8
 )
 
-type attrib struct {
-	prefix []byte
-	messageID byte
-	payload []byte
+type Message struct {
+	MsgID   uint8
```
Comment (ghost) [inline ops/handshake/message.go]: The type of MsgID should be messageID, don't you think ?

File: ops/handshake/message.go (hunk 2 above)
Comment (amanjaiswalofficial) [reply ops/handshake/message.go:33]: In first few cases like interested and notchoking, there isn't any payload. So currently Im just going with that. THough later I'll certainly add the option for it, as of now didn't seem the need for it. What do you think?

File: ops/handshake/message.go (hunk 3 above)
Comment (amanjaiswalofficial) [reply ops/handshake/message.go]: Got it. Will do so.

File: ops/handshake/message.go (hunk 2 above)
Comment (ghost) [reply ops/handshake/message.go:33]: Yeah, you're right, we don't need that too much right now, i just want to make sure everything is working well. All good for now.

**Review by ghost — APPROVED** [review]

**Pull request merged**